### DISCUSSION TITLE
How to get policies assigned to bucket using java sdk?

### DISCUSSION THREAD

**Discussion by tarekabouzeid** [minio/minio#14373] [Q&A] [answered]
Hi,

I have Minio configured to use LDAP, and I have created some policies to provide certain access to specific bucket, then I assign that policy to specific LDAP group or user.

policy is:
`
{
    "ID": "my_policy",
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "s3:ListBucket"
            ],
            "Resource": [
                "arn:aws:s3:::artifacts"
            ],
            "Condition": {
                "StringLike": {
                    "s3:prefix": [
                        "*"
                    ]
                }
            }
        }
    ]
}
`

I am trying to use the SDK to get the policies assigned to specific bucket, but i don't get anything, or from Minio UI, when i go to that bucket -> Access Audit, I don't see any policies assigned to it. 
Also from Minio UI, when i go to the policy the "Groups" and "Users" are grayed out and cannot be accessed in the new UI.

But the permissions works fine on bucket though.

This happens to all the buckets.

so is there a way to get the currently assigned policies for a specific bucket ?
What I am trying to achieve is, via the SDK , check what are the allowed actions for the currently used access key and secret key. 

Java SDK Version : 8.3.6
Minio Version : RELEASE.2022-02-16T05-54-01Z

Best Regards,

**Comment by harshavardhana** [comment, marked as answer]
You have to use Console UI or mc command

**Reply by tarekabouzeid** [reply]
Thanks for your fast answer, using the Console UI , when I go to the policy, the "users" and "groups" tabs are not accessible 
![image](https://user-images.githubusercontent.com/11487213/155009222-201866f9-92da-4e3b-bd4d-628268b098a4.png)

From the mc command, I can get the policies assigned to particular group or user
`
mc admin group info minio_env "CN=group1,OU=main,DC=example,DC=com"
`
, but I don't know how to do it the other way, just provide the bucket name and get who can do what. can you please provide an example command ?

Best Regards,

**Reply by harshavardhana** [reply]
`mc admin policy info alias/ your-policy-name`

**Reply by murpg** [reply]
Are you able to use the AWS CLI to assign policies to groups?

**Reply by harshavardhana** [reply]
AWS CLI does not implement MinIO Admin APIs.
